pkg/network/vmispec/interface.go

```diff
@@ -188,3 +188,19 @@ func HasBindingPluginDeviceInfo(iface v1.Interface, bindingPlugins map[string]v1
 	}
 	return false
 }
+
+// FilterInterfaceStatuses returns a filtered slice of interface statues by some criteria
+// f must not be nil
+func FilterInterfaceStatuses(
```

Comment:
since this function is "half" generic,  why don't we write it completely generic?  so that it wold take a `[]T` and a ` func(item T) bool` returning the filtered `[]T` 
Or do we not use generics in this project?

Reply:
> since this function is "half" generic, why don't we write it completely generic? so that it wold take a []T and a  func(item T) bool returning the filtered []T

That is an Interesting idea that might be worth exploring in a follow up - as it would probably generate a vibrant discussion (and we need to fix this flake ASAP).

> Or do we not use generics in this project?

There is at least one place in the code-base that uses generics:
 
https://github.com/kubevirt/kubevirt/blob/7b7a6fd86dbe57bbadcb09d25bdbf3be1d3291cf/pkg/pointer/pointer.go#L22-L24

Reply:
See this [function]( https://github.com/samber/lo?tab=readme-ov-file#filter) (  [source code](https://github.com/samber/lo/blob/407b62d3f12eece919463f556c798661f5aabbbf/slice.go#L12)).
I don't think that it's worth adding this lib to project dependencies just for this, though HCO [did it](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/go.mod#L29) for even less

Reply:
Please consider following example of `FilterInterfacesSpec` nameing in that file for better consistency, and rename the function to `FilterInterfacesStatus`.

In general I agree we can make use of generics for filtering in various places, I think it should be done on different PR.

Reply:
@ormergi Done.